app/infrastructure/git/commit_reporter.rb

```diff
@@ -0,0 +1,39 @@
+require 'git'
+
+
+module GitCommit
+  class CommitReporter
+
+    EMPTY_SHA = '4b825dc642cb6eb9a060e54bf8d69288fbee4904'
+
+    def initialize(gitrepo)
+      @local = gitrepo.local
+      @git = Git.open(@local.git_repo_path)
+    end
+
+    def commit(sha)
+      @git.log(sha)
+    end
+
+    def commits(since=nil)
+      commits = get_all_commits
+      commits = commits.since(since) unless since.nil?
+      commits
+    end
+
+    def empty_commit
+      @git.gcommit(EMPTY_SHA)
+    end
+
+    private
+
+    def get_all_commits
```

Comment:
```
n = 500
retrieved = 0
commits = []
until retrieved < n do
  commits = @git.log(n)
  retrieved = commits.length
end

commits
```

Reply:
check again if there is a way to ask git library how many commits there are total